### PR TITLE
Chore/improve docker setup

### PR DESCRIPTION
**Pull request by paulodiovani**
- Move to yarn (better packages lock file, faster install)

- Use Alpine Linux (610MB smaller image)

    Alpine (244MB) Vs Debian (857MB)

- Install NodeJS from distro packages
- RUN all in a single step
- Add `.dockerignore` file to prevent fat image
- Add everything needed to the image

    `docker-compose` is only necessary for development, the image is auto-sufficient and can be run with
    `docker run --rm -p 4567:4567 insiter.io`

### CODE REVIEW

**Comment by paulodiovani** [comment]
@fabricionaweb @iagodahlem 
I had a lot of issues trying to install NPM@5 on this image. Mostly on post installation while `npm install`. Is it really important to have the `package-lock.json` in this project?

My suggestions are:

1. Go back to Debian and keep the 800MB+ image
2. Move to Yarn (and use a `yarn.lock` instead)
3. Do not use a package lock (✔️ this is my choice, unless the site build really depends on node package versions)

Please, vote.

**Comment by iagodahlem** [comment]
@paulodiovani my choice is `2. Move to Yarn`. The site build is not really dependent on node packages but is always good to lock our dependencies.

Anyway, I don't care too much with this, so what you choose will be good to me.

**Review by fabricionaweb — APPROVED** [review]

**Review by iagodahlem — APPROVED** [review]

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+.bundle
+.cache
+.DS_Store
+.git
+.sass-cache
+docker-compose.override.yml
+docker-compose.yml
+node_modules/
```
Comment (fabricionaweb) [inline .dockerignore]: Add `yarn-error.log`

File: .gitignore [hunk 2]
```diff
@@ -5,5 +5,5 @@
 .tmp
 node_modules/
 build/
-yarn.lock
+package-lock.json
 docker-compose.override.yml
```
Comment (fabricionaweb) [inline .gitignore:9]: Add `yarn-error.log`

Comment (paulodiovani) [reply .gitignore:9]: Done

**Pull request merged**